source/opt/ir_context.cpp

```diff
@@ -183,5 +188,248 @@ void IRContext::KillNamesAndDecorates(Instruction* inst) {
+        SpvOpConvertUToF,
+        SpvOpUConvert,
+        SpvOpSConvert,
+        SpvOpFConvert,
+        SpvOpQuantizeToF16,
+        SpvOpBitcast,
+        SpvOpSNegate,
+        SpvOpFNegate,
+        SpvOpIAdd,
+        SpvOpFAdd,
+        SpvOpISub,
+        SpvOpFSub,
+        SpvOpIMul,
+        SpvOpFMul,
+        SpvOpUDiv,
+        SpvOpSDiv,
+        SpvOpFDiv,
+        SpvOpUMod,
+        SpvOpSRem,
+        SpvOpSMod,
+        SpvOpFRem,
+        SpvOpFMod,
+        SpvOpVectorTimesScalar,
+        SpvOpMatrixTimesScalar,
+        SpvOpVectorTimesMatrix,
+        SpvOpMatrixTimesVector,
+        SpvOpMatrixTimesMatrix,
+        SpvOpOuterProduct,
+        SpvOpDot,
+        SpvOpIAddCarry,
+        SpvOpISubBorrow,
+        SpvOpUMulExtended,
+        SpvOpSMulExtended,
+        SpvOpAny,
+        SpvOpAll,
+        SpvOpIsNan,
+        SpvOpIsInf,
+        SpvOpLogicalEqual,
+        SpvOpLogicalNotEqual,
+        SpvOpLogicalOr,
+        SpvOpLogicalAnd,
+        SpvOpLogicalNot,
+        SpvOpSelect,
+        SpvOpIEqual,
+        SpvOpINotEqual,
+        SpvOpUGreaterThan,
+        SpvOpSGreaterThan,
+        SpvOpUGreaterThanEqual,
+        SpvOpSGreaterThanEqual,
+        SpvOpULessThan,
+        SpvOpSLessThan,
+        SpvOpULessThanEqual,
+        SpvOpSLessThanEqual,
+        SpvOpFOrdEqual,
+        SpvOpFUnordEqual,
+        SpvOpFOrdNotEqual,
+        SpvOpFUnordNotEqual,
+        SpvOpFOrdLessThan,
+        SpvOpFUnordLessThan,
+        SpvOpFOrdGreaterThan,
+        SpvOpFUnordGreaterThan,
+        SpvOpFOrdLessThanEqual,
+        SpvOpFUnordLessThanEqual,
+        SpvOpFOrdGreaterThanEqual,
+        SpvOpFUnordGreaterThanEqual,
+        SpvOpShiftRightLogical,
+        SpvOpShiftRightArithmetic,
+        SpvOpShiftLeftLogical,
+        SpvOpBitwiseOr,
+        SpvOpBitwiseXor,
+        SpvOpBitwiseAnd,
+        SpvOpNot,
+        SpvOpBitFieldInsert,
+        SpvOpBitFieldSExtract,
+        SpvOpBitFieldUExtract,
+        SpvOpBitReverse,
+        SpvOpBitCount,
+        SpvOpDPdx,
+        SpvOpDPdy,
+        SpvOpFwidth,
+        SpvOpDPdxFine,
+        SpvOpDPdyFine,
+        SpvOpFwidthFine,
+        SpvOpDPdxCoarse,
+        SpvOpDPdyCoarse,
+        SpvOpFwidthCoarse,
+        SpvOpPhi,
+        SpvOpImageSparseSampleImplicitLod,
+        SpvOpImageSparseSampleExplicitLod,
+        SpvOpImageSparseSampleDrefImplicitLod,
+        SpvOpImageSparseSampleDrefExplicitLod,
+        SpvOpImageSparseSampleProjImplicitLod,
+        SpvOpImageSparseSampleProjExplicitLod,
+        SpvOpImageSparseSampleProjDrefImplicitLod,
+        SpvOpImageSparseSampleProjDrefExplicitLod,
+        SpvOpImageSparseFetch,
+        SpvOpImageSparseGather,
+        SpvOpImageSparseDrefGather,
+        SpvOpImageSparseTexelsResident,
+        SpvOpImageSparseRead,
+        SpvOpSizeOf
+        // TODO(dneto): Add instructions enabled by ImageQuery
+    });
+  }
+}
+
+void IRContext::AddCombinatorsForExtesnion(ir::Instruction* extension) {
```

Comment:
Typo "Extension" is misspelled.
I can fix that before merge.

Reply:
Fixed.